src/libraries/System.Private.CoreLib/src/System/Threading/ThreadInt64PersistentCounter.cs

```diff
@@ -2,79 +2,69 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Collections.Generic;
 using System.Diagnostics;
 using System.Runtime.CompilerServices;
+using Internal.Runtime.CompilerServices;
 
 namespace System.Threading
 {
     internal sealed class ThreadInt64PersistentCounter
     {
-        // This type is used by Monitor for lock contention counting, so can't use an object for a lock. Also it's preferable
-        // (though currently not required) to disallow/ignore thread interrupt for uses of this lock here. Using Lock directly
-        // is a possibility but maybe less compatible with other runtimes. Lock cases are relatively rare, static instance
-        // should be ok.
         private static readonly LowLevelLock s_lock = new LowLevelLock();
 
-        private readonly ThreadLocal<ThreadLocalNode> _threadLocalNode = new ThreadLocal<ThreadLocalNode>(trackAllValues: true);
         private long _overflowCount;
 
+        // This should be a HashSet, that type is currently not available in CoreLib
```

Comment:
This is not the case anymore. HashSet was moved to CoreLib.

Reply:
Updated to use HashSet